web/app/js/components/LineGraph.jsx

```diff
@@ -96,7 +108,15 @@ export default class LineGraph extends React.Component {
     this.updateAxes();
   }
 
-  updateGraph(){
+  updateGraph() {
+    if (_.isEmpty(this.props.data)) {
+      this.loadingMessage
+        .text("---")
+        .style("opacity", 1);
+    } else {
+      this.loadingMessage.style("opacity", 0);
+    }
+
```

Comment:
This logic looks correct to me, but in practice I'm seeing graphs that display both the timeseries and the loading message:

![screen shot 2017-12-13 at 10 53 11 am](https://user-images.githubusercontent.com/9226/33957699-3708d890-dff7-11e7-9874-0c4c6890e150.png)

Reply:
I think this is leftover state from whatever graph / data was there before.
I've added a check that removes this message, but the existing problem I think is that we're switching which 3 deployments we show, and the timeseries data sticks around when it shouldn't. I'm going to try to fix that separately.